terraform_alert_config/import_adapter/gke.tf

```diff
@@ -0,0 +1,108 @@
+# Google Kubernetes Engine prosessing
+# Kubernetes deployment Dicom Impor Adapter
+resource "kubernetes_deployment" "dicom-adapter" {
+  metadata {
+    name = "dicom-adapter"
+    namespace = "namespace-dicom-import-adapter"
+    labels = {
+      App = "dicom-adapter"
+    }
+  }
+
+  # customize nodes params 
+  spec {
+    replicas = var.replicas
+    selector {
+      match_labels = {
+        App = "dicom-adapter"
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          App = "dicom-adapter"
+        }
+      }
+      spec {
+        # Customize dicom import adapter settings
+        container {
+          image = var.image
+          name  = "dicom-import-adapter"
+
+          port {
+            container_port = var.dimse_port
+            protocol = "TCP"
+          }
+
+          resources {
+            limits {
+              cpu    = "0.5"
+              memory = "512Mi"
+            }
+            requests {
+              cpu    = "250m"
+              memory = "50Mi"
+            }
+          }
+
+          args = [
+           "--dimse_aet=IMPORTADAPTER",
+           "--dimse_port=${var.dimse_port}",
+           "--monitoring_project_id=${var.project_id}",
+           "--gcs_backup_project_id=${var.project_id}",
+           "--dicomweb_address=https://healthcare.googleapis.com/v1/projects/${var.project_id}/locations/${var.region}/datasets/${var.dataset}/dicomStores/${var.store}/dicomWeb",
+           "--persistent_file_upload_retry_amount=${var.persistent_file_upload_retry_amount}",
+           "--min_upload_delay=${var.min_upload_delay}",
+           "--max_waiting_time_btw_uploads=${var.max_waiting_time_btw_uploads}",
+           "--persistent_file_storage_location=${var.persistent_file_storage_location}",
+           "--oauth_scopes=https://www.googleapis.com/auth/cloud-healthcare",
```

Comment:
Don't we also need the oauth scope for writing to GCS for making backups?

Reply:
Now, to create a bucket and dataset, user credentials are used that are installed at the configuration step using gcloud auth login or specified in ENV. If necessary, we can add oauth scope for writing to GCS to the google_container_cluster resource or to the container arguments

Reply:
But for the adapter itself, writing to GCS to make a backup for a file, would it be able to write to GCS without an oauth scope? Were you able to test?